backend/app/server/templates.go

```diff
@@ -0,0 +1,297 @@
+package server
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/go-chi/chi/v5"
+	log "github.com/go-pkgz/lgr"
+	"github.com/pkg/errors"
+	"github.com/umputun/secrets/backend/app/messager"
+	"github.com/umputun/secrets/backend/app/store"
+	"github.com/umputun/secrets/backend/app/validator"
+	"github.com/umputun/secrets/backend/ui"
+	"html/template"
+	"io/fs"
+	"net/http"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	baseTmpl  = "base"
+	mainTmpl  = "main"
+	errorTmpl = "error"
+
+	msgKey     = "message"
+	pinKey     = "pin"
+	expKey     = "exp"
+	expUnitKey = "expUnit"
+	keyKey     = "key"
+)
+
+type createMsgForm struct {
+	Message string
+	Exp     int
+	MaxExp  string
+	ExpUnit string
+	validator.Validator
+}
+
+type showMsgForm struct {
+	Key     string
+	Message string
+	validator.Validator
+}
+
+type templateData struct {
+	Form    any
+	PinSize int
+}
+
+func (s Server) render(w http.ResponseWriter, status int, page, tmplName string, data any) {
+	ts, ok := s.TemplateCache[page]
+	if !ok {
+		err := fmt.Errorf("the template %s does not exist", page)
+		log.Printf("[ERROR] %v", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	buf := new(bytes.Buffer)
+
+	if tmplName == "" {
+		tmplName = baseTmpl
+	}
+	err := ts.ExecuteTemplate(buf, tmplName, data)
+	if err != nil {
+		log.Printf("[ERROR] %v", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(status)
+	_, err = buf.WriteTo(w)
+	if err != nil {
+		log.Printf("[ERROR] %v", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+}
+
+// renders the home page
```

Comment:
it would be nice to have comments to all controllers showing the url and describing query param/body if uses any